src/frontend/src/optimizer/plan_node/stream_vector_index_lookup_join.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pretty_xmlish::XmlNode;
+use risingwave_common::bail;
+use risingwave_pb::plan_common::PbVectorIndexReaderDesc;
+use risingwave_pb::stream_plan::PbVectorIndexLookupJoinNode;
+
+use crate::optimizer::plan_node::expr_visitable::ExprVisitable;
+use crate::optimizer::plan_node::generic::{PhysicalPlanRef, VectorIndexLookupJoin};
+use crate::optimizer::plan_node::stream::StreamPlanNodeMetadata;
+use crate::optimizer::plan_node::utils::{Distill, childless_record};
+use crate::optimizer::plan_node::{
+    ExprRewritable, PlanBase, PlanTreeNodeUnary, Stream, StreamNode, StreamPlanRef,
+};
+use crate::optimizer::property::StreamKind;
+use crate::stream_fragmenter::BuildFragmentGraphState;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct StreamVectorIndexLookupJoin {
+    pub base: PlanBase<Stream>,
+    pub core: VectorIndexLookupJoin<StreamPlanRef>,
+}
+
+impl StreamVectorIndexLookupJoin {
+    pub fn new(core: VectorIndexLookupJoin<StreamPlanRef>) -> crate::error::Result<Self> {
+        if core.input.stream_kind() != StreamKind::AppendOnly {
+            bail!("StreamVectorIndexLookupJoin only support append only input")
+        }
+        Ok(Self::with_core(core))
+    }
+
+    fn with_core(core: VectorIndexLookupJoin<StreamPlanRef>) -> Self {
+        assert_eq!(core.input.stream_kind(), StreamKind::AppendOnly);
+        let base = PlanBase::new_stream_with_core(
+            &core,
+            core.input.distribution().clone(),
+            core.input.stream_kind(),
+            core.input.emit_on_window_close(),
+            core.input.watermark_columns().clone(),
+            core.input.columns_monotonicity().clone(),
+        );
+        Self { base, core }
+    }
```

Comment:
I think we need to think about what's the `stream_key` of VectorIndexLookupJoin. Considering it is a join, so we can't use the input's stream_key as the output of the join stream key.

Reply:
Considering there is no exchange would be used to enforce the join input, the stream key of VectorIndexLookupJoin should be the left stream_key + right stream_key

Reply:
> Considering it is a join, so we can't use the input's stream_key as the output of the join stream key.

It only has the same name `join` as normal join, and all others are different. There is even no right stream_key, because it yields a single row for each input row, with the output row as `(original row + array(vector topn result))` so the stream key of the input row should be the same as the output row.